clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -764,7 +764,7 @@ public boolean refreshCommittedOffsetsIfNeeded(Timer timer) {
                                                                         final Timer timer) {
         if (partitions.isEmpty()) return Collections.emptyMap();
 
-        final Generation generation = generation();
+        final Generation generation = generationIfStable();
```

Comment:
What do you think about renaming this local `generation` variable (and the one below) to distinguish them from the `AbstractCoordinator` field (or vice versa)? To make it easier to keep track of which `generation` variables can be null (these) and which cannot (instance variable).
Not sure there's a better name out there, just something to consider.

Reply:
The `generation` variable here is used to construct the offset-commit request. I can rename it accordingly.